### ISSUE TITLE
Loss Value Too Big

### ISSUE THREAD

**Issue by guoyang9** [yoyololicon/pytorch-NMF#23]
Hi, I tried with a large sparse matrix, say 100, 000 \times 100, 000. When I trained with the vanilla NMF model, the losses are too big - 5e3.

Do you have any idea of what issues it could be?

Thank you for the contribution.

**Comment by guoyang9** [comment]
Thank you for your prompt response. 
I suspect the quality being not satisfactory is because that NMF cannot handle such sparse matrix (99.99% sparsity).

**Comment by thomas-jac** [comment]
Hey, @guoyang9! Did you end up figuring out a solution to this? I am also working with a big sparse matrix and running into the same issue. I also think one other source of the problem is the sensitivity to the initializations of W and H.
